core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -242,20 +242,20 @@ export class StateGraphLayoutService {
         depth: SENTINEL_DEPTH,
         offset: SENTINEL_OFFSET,
         reachable: false,
-        x0: null,
-        y0: null,
-        xLabel: null,
-        yLabel: null,
-        id: null,
-        label: null,
-        height: null,
-        width: null,
-        reachableFromEnd: false
+        x0: 0,
+        y0: 0,
+        xLabel: 0,
+        yLabel: 0,
+        id: '',
+        label: '',
+        height: 0,
+        width: 0,
+        reachableFromEnd: false,
```

Comment:
@vojtechjelinek  It doesn't really matter what we initialize here for these values since we are going to change them later on in BFS. **Refer Lines: 439 - 457**
![image](https://user-images.githubusercontent.com/75239620/144301609-5c8d48b7-620c-4191-a0a9-565720849062.png)

These values are never going to be null according to my understanding so I tried to avoid `null` as much as I can.

Reply:
It always matters. It might not matter now, but later some changes to the code might influence it. Ther preferred thing to do here is to get rid of filling the empty values. If that is not possible then we should use `null`s.

Reply:
Why do we need to wait for later? Can we assign the correct values when we create the object?

Reply:
@

> Why do we need to wait for later? Can we assign the correct values when we create the object?

No, actually we are doing some calculations on the input values and the other values of `nodeData` like `depth: ` and `offset: `, and then we are assigning them. We are not directly assigning them. If you see the whole function then the first place where we are using these `y0, x0, yLabel, xLabel, height, width` is here:

![image](https://user-images.githubusercontent.com/75239620/145208519-a1e2070e-9144-4edb-a18f-412bdbf834c6.png)

Here, we are initializing these values along with the `id` and `label`.

Reply:
But then we should use `null`, or you can somehow transform this so that you save these values into a different object and then create this object here.

Reply:
@vojtechjelinek yeah! But still, even if we create a different object it will be of interface `NodeData` and we will get a TS error because we can not assign `null` to a `number` and if I change `number`  to `number | null` in the interface itself then a hell lot of if-else statements will be required at various places to remove other TS error.

Reply:
@vojtechjelinek I am looking for a hack not to change the `interface NodeData` and add `null` there.